.github/actions/pre_cache/action.yml

```diff
@@ -0,0 +1,43 @@
+name: "Pre-Cache"
+runs:
+  using: composite
+  steps:
+    - name: Get 'make' checksum
+      run: |
+        cp -f /usr/bin/make make.bin
+        md5sum make.bin > $GITHUB_WORKSPACE/make.md5
+      shell: bash -ileo pipefail {0}
+    - name: Get 'dot' checksum
+      run: |
+        if [ -f /usr/bin/dot ] ; then
+          cp -f /usr/bin/dot dot.bin
+          md5sum dot.bin > $GITHUB_WORKSPACE/dot.md5
+        else
+          touch dot.bin
+          md5sum dot.bin > $GITHUB_WORKSPACE/dot.md5
+          rm -f dot.bin
```

Comment:
These lines specifically make me think that my theory that the `make`/`dot` caching+installation actions aren't actually doing anything in the case of a cache hit is true.

For `make`, I'd bet that nothing happens on cache hits and make isn't (re-)installed, but the binary is already installed in the GitHub actions images, so the `Get 'make' checksum` stage above works. If caching+installation of `dot` worked, then I don't think it should ever be possible for `/usr/bin/dot` to _not_ exist at this point, so the fact that the `else` clause had to be added to handle that suggests that `dot` just isn't installed or restored from the cache on cache hits.